HabboHotel/Groups/Forums/GroupForumSettings.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Dapper;
+using Plus.Database;
+using Plus.HabboHotel.GameClients;
+using System;
+using System.Data;
+using System.Threading.Tasks;
+
+namespace Plus.HabboHotel.Groups.Forums
+{
+    public class GroupForumSettings
+    {
+        public GroupForum ParentForum { get; private set; }
+
+        public int WhoCanRead;
+        public int WhoCanPost;
+        public int WhoCanInitDiscussions;
+        public int WhoCanModerate;
+        private readonly IDatabase _database;
+
+        public GroupForumSettings(IDatabase database, GroupForum Forum)
+        {
+            _database = database;
+            ParentForum = Forum;
+
+            using var connection = _database.Connection();
+
+            var getForumSettingsSQL = "SELECT * FROM group_forums_settings WHERE group_id = @id";
+            var forumSettings = connection.QueryFirstOrDefault(getForumSettingsSQL, new { id = Forum.Id });
+
+            if (forumSettings is null)
+            {
+                connection.Execute("REPLACE INTO group_forums_settings (group_id) VALUES (@id);SELECT * FROM group_forums_settings WHERE group_id = @id", new { id = Forum.Id });
+            }
+
+            WhoCanRead = forumSettings?.who_can_read;
+            WhoCanPost = forumSettings?.who_can_post;
+            WhoCanInitDiscussions = forumSettings?.who_can_init_discussions;
+            WhoCanModerate = forumSettings?.who_can_mod;
+        }
+
+        public async Task Save()
+        {
+            using var connection = _database.Connection();
+            await connection.ExecuteAsync("UPDATE group_forums_settings SET who_can_read = @a, who_can_post = @b, who_can_init_discussions = @c, who_can_mod = @d WHERE group_id = @id",
+                new { id = ParentForum.Id, a = WhoCanRead, b = WhoCanPost, c = WhoCanInitDiscussions, d = WhoCanModerate });
+        }
+
+        public GroupForumPermissionLevel GetLevel(int n)
```

Comment:
You can directly cast integers to an enum, so this method might not be needed.
Also could be static.